easybuild/easyblocks/j/java.py

```diff
@@ -27,13 +27,42 @@
 
 @author: Jens Timmerman (Ghent University)
 """
+
+import os
+import shutil
+
+from distutils.version import LooseVersion
 from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.tools.run import run_cmd
 
 
 class EB_Java(PackedBinary):
     """Support for installing Java as a packed binary file (.tar.gz)
     Use the PackedBinary easyblock and set some extra paths.
     """
+    
+    def extract_step(self):
+        """Unpack the source"""
+        if LooseVersion(self.version) < LooseVersion('1.7'):
+            cmd = 'cd %s; bash %s' % (self.builddir, self.src[0]['path'])
+            run_cmd(cmd, log_all=True, simple=True, inp='')
+        else:
+            PackedBinary.extract_step(self)
+    
+    def install_step(self):
+        if LooseVersion(self.version) < LooseVersion('1.7'):
+            src = self.builddir + '/' + 'jdk%s' % self.version
+            dst = self.installdir
+            for item in os.listdir(src):
+                s = os.path.join(src, item)
+                d = os.path.join(dst, item)
+                if os.path.isdir(s):
+                    shutil.copytree(s, d, symlinks=True, ignore=None)
+                else:
+                    shutil.copy2(s, d)
```

Comment:
@pescobar you can simplify this by removing the existing (empty) installdir and just use `copytree`:

```python
if LooseVersion(self.version) < LooseVersion('1.7'):
    try:
        os.rmdir(self.installdir)
        shutil.copytree(os.path.join(self.builddir, 'jdk%s' % self.version), self.installdir)
    except OSError, err:
        raise EasyBuildError("Failed to install by copying: %s", err)
```